packages/account/src/Sections/Profile/PhoneVerification/__test__/didnt-get-the-code-modal.spec.tsx

```diff
@@ -0,0 +1,114 @@
+import userEvent from '@testing-library/user-event';
+import { useGetPhoneNumberOTP } from '@deriv/hooks';
+import { VERIFICATION_SERVICES } from '@deriv/shared';
+
+jest.mock('@deriv/hooks', () => ({
+    ...jest.requireActual('@deriv/hooks'),
+    useGetPhoneNumberOTP: jest.fn(() => ({
+        requestOnWhatsApp: jest.fn(),
+        requestOnSMS: jest.fn(),
+    })),
+}));
+
+describe('DidntGetTheCodeModal', () => {
+    const mock_store = mockStore({});
+    const mockSetShouldShowDidntGetTheCodeModal = jest.fn();
+    const mockSetStartTimer = jest.fn();
+    const mockSetOtpVerification = jest.fn();
+
+    const renderComponent = (phone_verification_type: string) => {
+        render(
+            <StoreProvider store={mock_store}>
+                <DidntGetTheCodeModal
+                    should_show_didnt_get_the_code_modal={true}
+                    phone_verification_type={phone_verification_type}
+                    setShouldShowDidntGetTheCodeModal={mockSetShouldShowDidntGetTheCodeModal}
+                    setStartTimer={mockSetStartTimer}
+                    setOtpVerification={mockSetOtpVerification}
+                />
+            </StoreProvider>
+        );
+    };
+    it('should render DidntGetTheCodeModal', () => {
+        renderComponent(VERIFICATION_SERVICES.SMS);
+        expect(screen.getByText(/Get a new code/)).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Resend code/ })).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Send code via WhatsApp/ })).toBeInTheDocument();
+        expect(screen.getByText(/or/)).toBeInTheDocument();
+        expect(screen.getByRole('button', { name: /Change phone number/ })).toBeInTheDocument();
+    });
+
+    it('should show Send code via SMS if phone_verification_type is whatsapp', () => {
+        renderComponent(VERIFICATION_SERVICES.WHATSAPP);
+        expect(screen.getByRole('button', { name: /Send code via SMS/ })).toBeInTheDocument();
+    });
+
+    it('should render setOtpVerification and setShouldShowDidintGetTheCodeModal when Change phone number is clicked, should not render setStartTimer', () => {
+        renderComponent(VERIFICATION_SERVICES.SMS);
+        const change_phone_number_button = screen.getByRole('button', { name: /Change phone number/ });
+        userEvent.click(change_phone_number_button);
+        expect(mockSetShouldShowDidntGetTheCodeModal).toBeCalled();
+        expect(mockSetOtpVerification).toBeCalled();
+        expect(mockSetStartTimer).not.toBeCalled();
+    });
+
+    it('should render setStartTimer and setShouldShowDidintGetTheCodeModal when Resend code is clicked', () => {
+        renderComponent(VERIFICATION_SERVICES.SMS);
+        const resend_code_button = screen.getByRole('button', { name: /Resend code/ });
+        userEvent.click(resend_code_button);
+        expect(mockSetShouldShowDidntGetTheCodeModal).toBeCalled();
+        expect(mockSetStartTimer).toBeCalled();
```

Comment:
can we use .toHaveBeenCalledTimes() instead?
I think here in our tests we refere to the same mock, so if in first test we call the mock fn, in next test it will be already called before running the test even..